### PR TITLE
Add more scenarios

### PR DESCRIPTION
**Pull request by billwert**

### CODE REVIEW

**Comment by billwert** [comment]
this failure was due to the mismatch we where 'netcoreapp3.0' meant 'netcoreapp3.1', I think.

File: .gitignore [hunk 1]
```diff
@@ -52,6 +52,12 @@ bld/
 # Visual Studio 2017 auto generated files
 Generated\ Files/
 
+# scenario intermediate directories
+src/scenarios/*/pub
+src/scenarios/*/app
+src/scenarios/*/bin
+src/scenarios/*/traces
```
Comment (ooooolivia) [inline .gitignore]: add obj folder?

File: src/scenarios/emptyvbconsoletemplate/test.py [hunk 2]
```diff
@@ -5,8 +5,8 @@
 import os
 from shared.runner import TestTraits, Runner
 
-SCENARIONAME = '.NET Core 5.0 Console Template'
-EXENAME = '50consoletemplate'
+SCENARIONAME = 'Empty VB Console Template'
+EXENAME = 'emptycsconsoletemplate'
```
Comment (ooooolivia) [inline src/scenarios/emptyvbconsoletemplate/test.py]: did u mean ```emptyvbconsoletemplate```?

File: src/scenarios/init.ps1 [hunk 3]
```diff
@@ -1,2 +1,2 @@
-$scripts = Join-Path $PSScriptRoot '..\..\scripts\' -Resolve
+$scripts = Join-Path $PSScriptRoot '..\..\scripts' -Resolve
```
Comment (ooooolivia) [inline src/scenarios/init.ps1]: just wondering is it more like a nit thing to remove the slash or otherwise it would break

Comment (billwert) [reply src/scenarios/init.ps1]: just a nit thing. I don't think it breaks anything.

File: src/scenarios/emptyvbconsoletemplate/test.py (hunk 2 above)
Comment (billwert) [reply src/scenarios/emptyvbconsoletemplate/test.py]: I did! good catch.

File: .gitignore (hunk 1 above)
Comment (billwert) [reply .gitignore]: do we have anything that actually makes an obj folder in this place? I haven't seen one yet.

Comment (ooooolivia) [reply .gitignore]: oops  I meant bin and obj in the app folder and they are already covered, my mistake.

File: src/scenarios/shared/precommands.py [hunk 4]
```diff
@@ -50,16 +52,27 @@ def __init__(self):
         args = parser.parse_args()
         self.configuration = args.configuration
```
Comment (ooooolivia) [inline src/scenarios/shared/precommands.py]: It breaks here when operation is RESTORE because it doesnt have configuration as arg

Comment (billwert) [reply src/scenarios/shared/precommands.py]: Good catch! As discussed I'm just going to remove the `restore` verb.

File: src/scenarios/shared/runner.py [hunk 5]
```diff
@@ -104,5 +107,6 @@ def run(self):
                              iterationsetup=None,
                              setupargs=None,
                              workingdir=const.TMPDIR,
-                             environmentvariables=envlistbuild
+                             environmentvariables=envlistbuild,
+                             processwillexit=self.traits.processwillexit
```
Comment (ooooolivia) [inline src/scenarios/shared/runner.py]: should add ```measurementdelay=None```

File: src/scenarios/staticconsoletemplate/src/Program.cs [hunk 6]
```diff
@@ -1,6 +1,6 @@
 ﻿using System;
 
-namespace _50consoletemplate
+namespace staticconsoletemplate
```
Comment (ooooolivia) [inline src/scenarios/staticconsoletemplate/src/Program.cs]: is staticconsoletemplate same as netcoreapp2.0 src ?  i dont see any diff between them except their names

Comment (ooooolivia) [reply src/scenarios/staticconsoletemplate/src/Program.cs]: maybe we could just combine them?

File: src/scenarios/staticconsoletemplate/test.py [hunk 7]
```diff
@@ -0,0 +1,18 @@
+'''
+C# Console app
+'''
+import sys
+import os
```
Comment (ooooolivia) [inline src/scenarios/staticconsoletemplate/test.py]: these imports are not used. should we delete them?

File: src/scenarios/staticvbconsoletemplate/pre.py [hunk 8]
```diff
@@ -0,0 +1,14 @@
+'''
+pre-command
+'''
+import os
+import sys
+from performance.logger import setup_loggers
+from shared import const
+from shared.precommands import PreCommands
+from test import EXENAME
```
Comment (ooooolivia) [inline src/scenarios/staticvbconsoletemplate/pre.py]: same, EXENAME is not used.

File: src/tools/ScenarioMeasurement/Startup/GenericStartupParser.cs [hunk 9]
```diff
@@ -40,7 +40,7 @@ public IEnumerable<Counter> Parse(string mergeTraceFile, string processName, ILi
 
                 source.Kernel.ProcessStart += evt =>
                 {
-                    if (processName.Equals(evt.ProcessName, StringComparison.OrdinalIgnoreCase) && pids.Contains(evt.ProcessID) && evt.CommandLine == commandLine)
+                    if (processName.Equals(evt.ProcessName, StringComparison.OrdinalIgnoreCase) && pids.Contains(evt.ProcessID) && evt.CommandLine.Trim() == commandLine.Trim())
```
Comment (ooooolivia) [inline src/tools/ScenarioMeasurement/Startup/GenericStartupParser.cs]: Would it break here without trim?  If so, can we do trim in Startup.cs so we don't need to add this code to every parser?

Comment (billwert) [reply src/tools/ScenarioMeasurement/Startup/GenericStartupParser.cs]: The erroneous space is on the event payload, so it should be here. I'm trimming both out of paranoia. I'll clean this up when I refactor this code.

File: src/scenarios/staticconsoletemplate/src/Program.cs (hunk 6 above)
Comment (billwert) [reply src/scenarios/staticconsoletemplate/src/Program.cs]: sure looks like it. I'll leave it to you in the SDK test refactoring.

File: src/scenarios/staticconsoletemplate/test.py (hunk 7 above)
Comment (billwert) [reply src/scenarios/staticconsoletemplate/test.py]: done

File: src/scenarios/staticvbconsoletemplate/pre.py (hunk 8 above)
Comment (billwert) [reply src/scenarios/staticvbconsoletemplate/pre.py]: made a change to use it.

File: src/tools/ScenarioMeasurement/Startup/Startup.cs [hunk 10]
```diff
@@ -215,6 +215,10 @@ static void checkArg(string arg, string name)
                 }
                 TraceEventSession.Merge(files.ToArray(), traceFileName);
 
+                if (guiApp)
+                {
+                    appExe = Path.Join(workingDir, appExe);
```
Comment (ooooolivia) [inline src/tools/ScenarioMeasurement/Startup/Startup.cs]: wondering why we need to join the paths here; seems evt.commandLine only takes whatever appExe is

Comment (billwert) [reply src/tools/ScenarioMeasurement/Startup/Startup.cs]: in the GUI app case the event payload includes the full path to the exe. We happen to know (by convention) that the specified working dir is always the root to the app to run. We could change this to always pass in the full path also if this becomes problematic.

Comment (ooooolivia) [reply src/tools/ScenarioMeasurement/Startup/Startup.cs]: "We happen to know (by convention) that the specified working dir is always the root to the app to run." not true for sdk tests but it's fine as long as guiApp=false for sdk tests...

Comment (billwert) [reply src/tools/ScenarioMeasurement/Startup/Startup.cs]: fair point. though maybe we should rethink that. if SDK tests are so different we'll have trouble sharing templates etc..

**Pull request merged**